File: avalon-modules/avalon-authorization/src/main/java/io/github/lishangbu/avalon/authorization/entity/Permission.java
package io.github.lishangbu.avalon.authorization.entity;

import java.io.Serial;
import java.io.Serializable;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

/**
 * 权限(Permission)实体类
 *
 * @author lishangbu
 * @since 2025/08/28
 */
@Data
@Table(name = "[permission]")
public class Permission implements Serializable {
  @Serial private static final long serialVersionUID = 1L;

  /** 主键 */
  @Id private Long id;

  /** 权限名称 */
  private String name;

  /** 权限编码 */
  private String code;

  /** 父权限ID */
  private Long parentId;

  /** 请求方法 */
  private String method;

  /** 描述 */
  private String description;

  /** 是否启用 */
  private Boolean enabled;

  /** 排序顺序 */
  private Integer sortOrder;
}

File: modules/identity-access/src/main/kotlin/io/github/lishangbu/avalon/identity/access/domain/iam/Permission.kt
package io.github.lishangbu.avalon.identity.access.domain.iam


/**
 * 权限聚合的领域视图。
 *
 * @property id 权限标识。
 * @property menuId 所属菜单标识。
 * @property code 权限业务编码。
 * @property name 权限展示名称。
 * @property enabled 权限是否启用。
 * @property sortingOrder 权限排序值。
 * @property version 乐观锁版本号。
 */
data class Permission(
    val id: PermissionId,
    val menuId: MenuId,
    val code: String,
    val name: String,
    val enabled: Boolean,
    val sortingOrder: Int,
    val version: Long,
)